test/test_mkldnn.py

```diff
@@ -1510,12 +1520,18 @@ def test_lstm(self):
 
                 model1 = copy.deepcopy(model)
                 model2 = copy.deepcopy(model)
-                with torch.cpu.amp.autocast(enabled=bf16, dtype=torch.bfloat16), torch.no_grad() if not training else nullcontext():
+                with torch.amp.autocast(
+                    "cpu", enabled=bf16 or fp16, dtype=lowp_dtype
+                ), (torch.no_grad() if not training else nullcontext()):
                     with torch.backends.mkldnn.flags(enabled=False):
                         torch.manual_seed(seed)
-                        output1, (hn1, cn1) = self._cast_dtype(model1, bf16)(self._cast_dtype(input1, bf16),
-                                                                             (self._cast_dtype(h1, bf16),
-                                                                             self._cast_dtype(c1, bf16)))
+                        output1, (hn1, cn1) = self._cast_dtype(model1, bf16, fp16)(
+                            self._cast_dtype(input1, bf16, fp16),
+                            (
+                                self._cast_dtype(h1, bf16, fp16),
+                                self._cast_dtype(c1, bf16, fp16),
```

Comment:
Update according to _cast_dtype API change.

Reply:
Thanks for your comments. Revised.